### DISCUSSION TITLE
volume data user recommandation ?

### DISCUSSION THREAD

**Discussion by dcpc007** [felddy/foundryvtt-docker#214] [Q&A]
hi, in complement to several discussions about the access to data in the foundryvtt volume with FOUNDRY_UID and FOUNDRY_GID, i would know how you manage it more deeply.

My current case without near any custo : 
data is in /srv/foundryTT : 
- 421:421
- perm 755/644

i work with a local user for all the tasks on my server.
Then currently no write access on anything except sudo/root : nont realy a good conifg :)

The solution i see in discussions is : force UID/GID.
But i'm not sure it's a really good config to force on my local user UID/GID ?
I was thinking about create a user/group named "foundry:foundry", and put myself in the group, but then, still only read access due to 755 default.

Suggestion : do you think it's possible to change for 775/664 (and maybe better 770/660 !!), with SetGID on to preserve the group if needed ?

Otherwise, the precedent case will be : 
- all data belongs to foundry:foundry as "admin rw mode"
- local user include foundry group and then gains read only access to all
- need to switch to foundry user to manage data like delete, rename, create backup,... (with sudo / su)

Hope it can help :)

**Comment by felddy** [comment]
I'm not entirely sure what your question is, but I can explain a bit about the permissions the container uses.  

By default it will run using its internal `foundry` user and group witch is `421:421`.  You can override this with the `FOUNDRY_UID` and `FOUNDRY_GID` environment variables.  I do this on my instance so that it matches my user's UID:GID.  

You can set the ACLs of the files to whatever works for you.  The container will not modify the file mode bits.  It will `chown` the data directory to match the UID:GID that it is using unless you specify `CONTAINER_PRESERVE_OWNER`.  See: https://github.com/felddy/foundryvtt-docker#optional

I think it comes down to personal preference, and the scheme you are using on your server.

**Reply by dcpc007** [reply]
yes, i only try to put a solution, with a dedicated local user (better than the self user taht do others thing on the server (security recomandation).

Can we change the umask by default to create files in 660/770 by default ?

**Reply by felddy** [reply]
Good idea.  I'll look into adding a `CONTAINER_UMASK` variable that can be configured if needed.

**Reply by eltariel** [reply]
I found myself crashing into these kind of permission issues when I tried to update some excessively large shared asset packs from the host rather than inside the container. I bashed together a PR (#1325)  that implements `CONTAINER_UMASK` rather than sleeping last night...

Also, oops this is an old thread :rofl:
